src/pages/settings.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { setApiKey, getApiKey } from "../helpers/store";
+import { MODELS_API } from "../constants/API";
+import Header from "../components/Header";
+import { DownArrow, DownChevron } from "../svg";
+
+const SettingsPage = () => {
+  const [apiKey, setApiKeyState] = useState("");
+  const [selectedModel, setSelectedModel] = useState("");
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function fetchSettings() {
+      const key = await getApiKey();
+      setApiKeyState(key || "");
+      // Fetch and set the selected model
+    }
+
+    fetchSettings();
+  }, []);
+
+  const handleSave = async () => {
+    await setApiKey(apiKey);
+    // Save the selected model
+    navigate("/");
```

Comment:
you'd need to save the model to the store as well, since it'll get wiped the moment you leave the settings screen.

Similar to `setApiKey` in `helpers/store.ts`, create another function to store the `model` in the store.